R/data.table.R

```diff
@@ -1881,17 +1881,64 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
 #    x
 #}
 
-
-as.matrix.data.table <- function(x,...)
-{
-  dm <- dim(x)
-  cn <- names(x)
+as.matrix.data.table <- function(x, rownames, ...) {
+  rn <- NULL
+  rnc <- NULL
+  if (!missing(rownames)) { # Convert rownames to a column index if possible
+    if (is.null(rownames)) {
+      warning("rownames is NULL, ignoring rownames")
+    } else if (length(rownames) == nrow(x)) {
```

Comment:
i think `length(rownames)` is called enough times (3) to warrant assigning it, perhaps here as `if (n_rn <- length(rownames))` or analogous

Reply:
It's good that you check vs. `nrow(x)` first for the `nrow(x) == 1L` case; please be sure to document behavior in this case.